src/lit-element.ts

```diff
@@ -11,286 +11,51 @@
-   * reference to a static node which must be directly manipulated.
-   * This should not be commonly needed. For tasks which should be performed
-   * before first render, use the element constructor.
-   */
-  _firstRendered() {}
-
-  /**
-   * Implement to customize where the element's template is rendered by
-   * returning an element into which to render. By default this creates
-   * a shadowRoot for the element. To render into the element's childNodes,
-   * return `this`.
-   * @returns {Element|DocumentFragment} Returns a node into which to render.
-   */
-  protected _createRoot(): Element|DocumentFragment {
-    return this.attachShadow({mode : 'open'});
-  }
-
-  /**
-   * Override which returns the value of `_shouldRender` which users
-   * should implement to control rendering. If this method returns false,
-   * _propertiesChanged will not be called and no rendering will occur even
-   * if property values change or `requestRender` is called.
-   * @param _props Current element properties
-   * @param _changedProps Changing element properties
-   * @param _prevProps Previous element properties
-   * @returns {boolean} Default implementation always returns true.
-   */
-  _shouldPropertiesChange(_props: object, _changedProps: object,
-                          _prevProps: object): boolean {
-    const shouldRender = this._shouldRender(_props, _changedProps, _prevProps);
-    if (!shouldRender && this.__resolveRenderComplete) {
-      this.__resolveRenderComplete(false);
-    }
-    return shouldRender;
-  }
+export abstract class LitElement extends UpdatingElement {
 
+  private _firstRendered = false;
   /**
-   * Implement to control if rendering should occur when property values
-   * change or `requestRender` is called. By default, this method always
-   * returns true, but this can be customized as an optimization to avoid
-   * rendering work when changes occur which should not be rendered.
-   * @param _props Current element properties
-   * @param _changedProps Changing element properties
-   * @param _prevProps Previous element properties
-   * @returns {boolean} Default implementation always returns true.
+   * Render method used to render the lit-html TemplateResult to the element's DOM.
+   * @param {TemplateResult} Template to render.
+   * @param {Element|DocumentFragment} Node into which to render.
+   * @param {String} Element name.
    */
-  protected _shouldRender(_props: object, _changedProps: object,
-                          _prevProps: object): boolean {
-    return true;
-  }
+  static render = render;
 
   /**
-   * Override which performs element rendering by calling
-   * `_render`, `_applyRender`, and finally `_didRender`.
-   * @param props Current element properties
-   * @param changedProps Changing element properties
-   * @param prevProps Previous element properties
+   * Override which performs element rendering by calling the `render` method.
```

Comment:
These will be the docs shown in the doc viewer. I don't think "Override which..." provides a lot of information. I'd copy a lot of the docs from UpdatingElement